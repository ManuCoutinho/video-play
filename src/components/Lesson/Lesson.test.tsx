import '@testing-library/jest-dom'
import { describe, it, vi } from 'vitest'
import { screen, render } from '@testing-library/react'
import { Lesson } from '.'

const lessonMock = {
  title: 'test title',
  duration: '04:33',
  onPlay: vi.fn(),
  isCurrent: false
}
describe('Name of the group', () => {
  it('should render the component with data', () => {
    render(<Lesson {...lessonMock} />)
    const button = screen.getByRole('button', { name: /test title 04:33/i })
    expect(button).toHaveAttribute('data-active', 'false')
  })
  it('should render the component without data', () => {
    render(<Lesson {...lessonMock} isCurrent={true} />)
    const button = screen.getByRole('button', { name: /test title 04:33/i })
    expect(button).toHaveAttribute('data-active', 'true')
  })
  it('should match to snapshot', () => {
    const { container } = render(<Lesson {...lessonMock} isCurrent={true} />)
    expect(container).toMatchSnapshot()
  })
})
