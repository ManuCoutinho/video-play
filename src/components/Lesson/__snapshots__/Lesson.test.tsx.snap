// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Name of the group > should match to snapshot 1`] = `
<div>
  <button
    class="flex items-center gap-3 text-sm text-info data-[active=true]:text-emerald-300 enabled:hover:text-emerald-200 px-4 py-2"
    data-active="true"
    disabled=""
  >
    <svg
      class="text-emerald-300 w-4 h-4"
      fill="currentColor"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128,26A102,102,0,1,0,230,128,102.12,102.12,0,0,0,128,26Zm0,192a90,90,0,1,1,90-90A90.1,90.1,0,0,1,128,218Zm35.33-95-48-32A6,6,0,0,0,106,96v64a6,6,0,0,0,9.33,5l48-32a6,6,0,0,0,0-10ZM118,148.79V107.21L149.18,128Z"
      />
    </svg>
    <span>
      test title
    </span>
    <span
      class="ml-auto font-mono text-xs text-secondary"
    >
      04:33
    </span>
  </button>
</div>
`;
